src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -6096,96 +6096,135 @@ void ObjectMgr::LoadAccessRequirements()
-                ar->item2 = 0;
-            }
-        }
+                uint8 requirement_type       = progression_requirement_row[0].GetUInt8();
+                uint32 requirement_id        = progression_requirement_row[1].GetUInt32();
+                std::string requirement_hint = progression_requirement_row[2].GetString();
+                uint8 requirement_faction    = progression_requirement_row[3].GetUInt8();
 
-        if (ar->quest_A)
-        {
-            if (!GetQuestTemplate(ar->quest_A))
-            {
-                sLog->outErrorDb("Required Alliance Quest %u not exist for map %u difficulty %u, remove quest done requirement.", ar->quest_A, mapid, difficulty);
-                ar->quest_A = 0;
-            }
-        }
+                ProgressionRequirement* progression_requirement = new ProgressionRequirement();
+                progression_requirement->id           = requirement_id;
+                progression_requirement->hint         = requirement_hint;
+                progression_requirement->faction      = (TeamId)requirement_faction;
 
-        if (ar->quest_H)
-        {
-            if (!GetQuestTemplate(ar->quest_H))
-            {
-                sLog->outErrorDb("Required Horde Quest %u not exist for map %u difficulty %u, remove quest done requirement.", ar->quest_H, mapid, difficulty);
-                ar->quest_H = 0;
-            }
-        }
+                switch (requirement_type)
+                {
+                case 0:
+                {
+                    //Achievement
+                    if (!sAchievementStore.LookupEntry(progression_requirement->id))
+                    {
+                        sLog->outErrorDb("Required achievement %u for faction %u not exist for map %u difficulty %u, remove or fix achievement requirement.", progression_requirement->id, requirement_faction, mapid, difficulty);
+                        delete progression_requirement;
+                        break;
+                    }
 
-        if (ar->achievement)
-        {
-            if (!sAchievementStore.LookupEntry(ar->achievement))
-            {
-                sLog->outErrorDb("Required Achievement %u not exist for map %u difficulty %u, remove quest done requirement.", ar->achievement, mapid, difficulty);
-                ar->achievement = 0;
-            }
+                    ar->achievements.push_back(progression_requirement);
+                    break;
+                }
+                case 1:
+                {
+                    //Quest
+                    if (!GetQuestTemplate(progression_requirement->id))
+                    {
+                        sLog->outErrorDb("Required quest %u for faction %u does not exist for map %u difficulty %u, remove or fix quest requirement.", progression_requirement->id, requirement_faction, mapid, difficulty);
+                        progression_requirement->id = 0;
+                    }
```

Comment:
Missing `                         delete progression_requirement;` ??? (look at the other cases)